File: app/src/main/java/com/mss/instamat/repositories/files/FilesHelper.java
package com.mss.instamat.repositories.files;

import android.graphics.Bitmap;
import android.os.Environment;
import android.support.annotation.NonNull;

import com.mss.instamat.domain.models.ImageInfo;

import org.jetbrains.annotations.NotNull;

import java.io.File;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.OutputStream;

import timber.log.Timber;

public class FilesHelper {

    private static final String SAVE_FOLDER = "instamat";

    @NotNull
    public String getFolderPath() {
        return getPublicDirectory() + File.separator + SAVE_FOLDER;
    }

    @NotNull
    public String getFilePath(String folder, @NonNull ImageInfo imageInfo) {
        return folder + File.separator + imageInfo.getId() + ".jpg";
    }

    public void createFolderIfNotExist(@NonNull String folder) {
        new File(folder).mkdirs();
    }

    @NotNull
    public OutputStream getFileOutputStream(String filePath) throws IOException {
        return new FileOutputStream(filePath);
    }

    public void deleteFileIfExist(String filePath) {
        File file = new File(filePath);
        if (file.exists()) {
            Timber.d("Delete file in %s", filePath);
            file.delete();
        }
    }

    public String getPublicDirectory() {
        return Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_PICTURES).getAbsolutePath();
    }

    public void saveBitmapToStream(@NonNull Bitmap bitmap, OutputStream outStream) throws IOException {
        bitmap.compress(Bitmap.CompressFormat.JPEG, 100, outStream);
        outStream.flush();
        outStream.close();
    }
}

File: app/src/main/java/com/mss/imagesearcher/repositories/files/FilesHelper.kt
package com.mss.imagesearcher.repositories.files

import android.graphics.Bitmap
import android.os.Environment
import com.mss.imagesearcher.domain.models.ImageInfo
import timber.log.Timber
import java.io.File
import java.io.FileOutputStream
import java.io.IOException
import java.io.OutputStream

class FilesHelper {

    companion object {
        private const val SAVE_FOLDER = "imagesearcher"
    }

    val folderPath: String
        get() = publicDirectory + File.separator + SAVE_FOLDER

    val publicDirectory: String
        get() = Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_PICTURES).absolutePath

    fun getFilePath(folder: String, imageInfo: ImageInfo): String {
        return folder + File.separator + imageInfo.id + ".jpg"
    }

    fun createFolderIfNotExist(folder: String) {
        File(folder).mkdirs()
    }

    @Throws(IOException::class)
    fun getFileOutputStream(filePath: String): OutputStream {
        return FileOutputStream(filePath)
    }

    fun deleteFileIfExist(filePath: String) {
        val file = File(filePath)
        if (file.exists()) {
            Timber.d("Delete file in %s", filePath)
            file.delete()
        }
    }

    @Throws(IOException::class)
    fun saveBitmapToStream(bitmap: Bitmap, outStream: OutputStream) {
        bitmap.compress(Bitmap.CompressFormat.JPEG, 100, outStream)
        outStream.flush()
        outStream.close()
    }
}
